File: C++/389.cpp
__________________________________________________________________________________________________
sample 4 ms submission
class Solution {
public:
    char findTheDifference(string s, string t) {
        char res = 0;
        for (char c : s) res ^= c;
        for (char c : t) res ^= c;
        return res;
    }
};
__________________________________________________________________________________________________
sample 8888 kb submission
auto fast_io = []()
{
    std::ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return nullptr;
}();

class Solution {
public:
    char findTheDifference(string s, string t) {
        char flag[26] = {0};
        for (auto i:s) ++flag[i - 'a'];
        for (auto i:t) {
            --flag[i - 'a'];
            if (flag[i - 'a'] < 0) return i;
        }
        
        return -1;
    }
};
__________________________________________________________________________________________________

File: Python3/389.py
__________________________________________________________________________________________________
sample 24 ms submission
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        import collections
        c_s = collections.Counter(s)
        c_t = collections.Counter(t)
        
        key_s = set(c_s.keys())
        key_t = set(c_t.keys())
        
        if len(key_t-key_s):
            return list(key_t-key_s)[0]
        
        else:
            for ks in key_s:
                if c_s[ks] != c_t[ks]:
                    return ks
__________________________________________________________________________________________________
sample 12960 kb submission
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        code = 0
        for ch in s+t:
            code ^= ord(ch)
        return chr(code)        
__________________________________________________________________________________________________
